### ISSUE TITLE
Problem: no reusable template for switching out data sources for lists with reactive filter

### ISSUE THREAD

**Issue by janat08** [Blockrazor/blockrazor#873]
Problem: we filter lists reactively in typeahead, problems, and home route, in typeahead and home route with just load the whole collection into local one or try to use it if it exists before which we still use a subscription for fast-render. In future, we might refractor this to save to browser memory and try to load from that and update it every so often or place an observer to maintain reactivity without querying DB with every keypress.

Solution: begin rendering lists where filtering is reactive (queries DB on every keypress on filter field) through a separate template that handles data with that in mind. Collections with corresponding local collections and method call to populate the local one could hold these properties the online collection.

**Comment by janat08** [comment]
https://atmospherejs.com/dburles/mongo-collection-instances
https://atmospherejs.com/dburles/collection-helpers

I could probably get away from the issue of template hierarchies instead introducing another helper that will either returns straight from DB or via local collection.

**Comment by janat08** [comment]
heh, I did it by creating a class.
proof:
```
  this.autorun(()=>{
    Currencies.localFind()
    Tracker.nonreactive(()=>{
      console.log(Currencies.local.find().count())
    })
  })
```

**Comment by janat08** [comment]
/problems uses Problems.find everywhere instead of currentData() and such. I could default any query to local collection or make it an option, or do it if local record exists, and then go on to do observeChanges if query is being used to update local collection. I obviusly could go through the codebase updating queries instead.

**Comment by janat08** [comment]
@gazhayes should I go and update things by hand.

**Comment by gazhayes** [comment]
To be honest, I'm a bit lost here now. Can you summarize the current problem?

**Comment by janat08** [comment]
NM.
